### PR TITLE
New version: Youdao.youdaodict 10.0.12.0

### PR DESCRIPTION
**Pull request by lychichem**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/118939)

### CODE REVIEW

**Comment by stephengillie** [comment]
> Manifest Error: Schema Error. [Schema validation failed. 
Error context: <root>[Installers][0][InstallerSha256] Description: Failed to match regex specified by 'pattern' constraint. 
Error context: <root>[Installers][0] Description: Failed to validate against schema associated with property name 'InstallerSha256'. 
Error context: <root>[Installers] Description: Failed to validate item #0 in array. 
Error context: <root> Description: Failed to validate against schema associated with property name 'Installers'. ] File: Youdao.YoudaoDict.installer.yaml 
Manifest Error: Failed to process field. [InstallerSha256] 
Manifest Error: Required field missing. [InstallerSha256]

File: manifests/y/Youdao/YoudaoDict/10.0.12.0/Youdao.YoudaoDict.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+PackageVersion: 10.0.12.0
+MinimumOSVersion: 10.0.0.0
+InstallerType: exe
+InstallerSwitches:
+  Silent: /S
+  SilentWithProgress: /S
+Installers:
+- Architecture: x86
+  InstallerUrl: https://download.ydstatic.com/cidian/static/10.0/20230904/YoudaoDictSetup.exe
+  InstallerSha256: 922302C9BABCB55AC546017E52F1190EA85B9AAE6AEA80E340C4830C744A7C2
```
Comment (stephengillie) [inline manifests/y/Youdao/YoudaoDict/10.0.12.0/Youdao.YoudaoDict.installer.yaml:14]:
```suggestion
  InstallerSha256: 922302C9BABCB555AC546017E52F1190EA85B9AAE6AEA80E340C4830C744A7C2
```

Comment (stephengillie) [reply manifests/y/Youdao/YoudaoDict/10.0.12.0/Youdao.YoudaoDict.installer.yaml:14]: Updated hash with downloaded file. I think the original hash wasn't long enough.

Comment (lychichem) [reply manifests/y/Youdao/YoudaoDict/10.0.12.0/Youdao.YoudaoDict.installer.yaml:14]: you are right. The reason is I cannot paste hash text after copying files in github.dev.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**